### PR TITLE
Feature/insights remote lib

### PR DESCRIPTION
**Pull request by bomoko**
This PR will add an "in cluster" gatherer for the new insights remote webservice

### CODE REVIEW

File: cmd/gatherInCluster.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		if dryRun {
+			if facts != nil {
+				log.Println("---- Dry run ----")
+				log.Printf("Would post the follow facts to '%s:%s'", projectName, environmentName)
+				s, _ := json.MarshalIndent(facts, "", "\t")
+				log.Println(string(s))
+			}
+		}
+	},
+}
```
Comment (timclifford) [inline cmd/gatherInCluster.go:94]: Could move some of this logic into separate functions and then call like:
```
tokenValue, err := getTokenValue()
if err != nil {
	log.Fatal(err)
}

...

gatherAndWriteFacts(gathererTypeArg, tokenValue)
```

File: cmd/gatherInCluster.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+// gatherCmd represents the gather command
+var gatherInClusterCmd = &cobra.Command{
+	Use:   "gather-in-cluster",
+	Short: "Running this command will invoke the registered gatherers in cluster",
+	Long:  `Running all the registered gatherers will inspect the system and write FACT data back to the Lagoon insights system via insights-remote`,
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
+		//get the basic env vars
+		if argStatic && argDynamic {
+			log.Fatalf("Cannot use both 'static' and 'dynamic' only gatherers - exiting")
+			os.Exit(1)
```
Comment (timclifford) [inline cmd/gatherInCluster.go:26]: No need for a os.Exit with the log.FatalF right?

File: gatherers/FilledFileGatherer.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+func (p *filledFileGatherer) GetGathererCmdType() string {
+	return GATHERER_TYPE_STATIC
+}
+
+func (p *filledFileGatherer) AppliesToEnvironment() bool {
+
+	applies := false
+
+	// Look into /tmp/facts for any .json file
+	filepath.Glob("/tmp/facts/*.json")
```
Comment (timclifford) [inline gatherers/FilledFileGatherer.go:22]: Do we need some error handling here i.e.:
```
	files, err := filepath.Glob("/tmp/facts/*.json")
	if err != nil {
		// Handle the error 
		return false
	}
```
Then if there are any .json files:
```
return len(files) > 0
```

File: gatherers/insights-remote-writer.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+			Value:       fact.Value,
+			Source:      fact.Source,
+			Description: fact.Description,
+			Category:    string(fact.Category),
+		}
+
+		insightsRemoteFacts.Facts = append(insightsRemoteFacts.Facts, f)
+	}
+
+	bodyString, _ := json.Marshal(insightsRemoteFacts)
```
Comment (timclifford) [inline gatherers/insights-remote-writer.go:32]: might want to have some error handling to this in a few places below to
```
bodyString, err := json.Marshal(insightsRemoteFacts)
if err != nil {
   return fmt.Errorf("failed to marshal insights remote facts: %w", err)
}
```

File: gatherers/insights-remote-writer.go [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+	fmt.Printf("Sending %v fact(s) to insights core\n", len(facts))
+
+	serviceEndpoint := viper.GetString("insights-remote-endpoint")
+	req, _ := http.NewRequest(http.MethodPost, serviceEndpoint, bytes.NewBuffer(bodyString))
+	req.Header.Set("Authorization", token)
+	req.Header.Set("Content-Type", "application/json")
+	client := &http.Client{}
+	response, err := client.Do(req)
```
Comment (timclifford) [inline gatherers/insights-remote-writer.go:41]: could add a defer here to:
```
defer response.Body.Close()
```

**Review by timclifford — COMMENTED** [review]
Looks good

File: cmd/gatherInCluster.go (hunk 1 above)
Comment (bomoko) [reply cmd/gatherInCluster.go:93]: Yeah, that's clean - I've been trying to keep things simpler and _not_ introducing functions prematurely (that is, where applicable, I want to keep the actual logic in one place, if it's not too complex).

So I'll refactor the get token stuff, but maybe leave the rest till it the function gets a little more complicated.

**Pull request merged**